File: Source/JapaneseLanguageTools.Contracts/Models/Requests/Base/ExportRequest.Model.cs
using JapaneseLanguageTools.Contracts.Enumerations;

namespace JapaneseLanguageTools.Contracts.Models.Requests.Base;

public class ExportRequestModel
{
    public required SnapshotType SnapshotType { get; set; }

    public required SnapshotFileFormat SnapshotFileFormat { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/requests/base/ExportRequest.Model.ts
import SnapshotFileFormat from "@/enumerations/SnapshotFileFormat";
import SnapshotType from "@/enumerations/SnapshotType";

export default interface ExportRequestModel {
  snapshotType: SnapshotType;
  snapshotFileFormat: SnapshotFileFormat;
}
